federation/pkg/federation-controller/util/meta.go

```diff
@@ -92,3 +98,33 @@ func DeepCopyApiTypeOrPanic(item interface{}) interface{} {
 	}
 	return result
 }
+
+func IsFederationOnlyObject(object runtime.Object) bool {
```

Comment:
As per #44139, it's possible to manipulate the annotations for an object in a type-agnostic way:

```golang
// import "k8s.io/apimachinery/pkg/api/meta"
accessor, err := meta.Accessor(obj)
if err != nil {
    ...
}
annotations := accessor.GetAnnotations()
..
accessor.SetAnnotations(annotations)
```